### ISSUE TITLE
NFS HP2 (2002) transparent cars

### ISSUE THREAD

**Issue by spartaque12** [crosire/d3d8to9#55]
hey crosire , d3d8to9 doesnt helped me to fix dat bug , i tried all cracks and versions of games , still transparent cars :/

https://imgur.com/a/ErO9r

**Comment by elishacloud** [comment]
This is happening because of an issue with the VertexShader.  The log file is showing this error:

```text
D:\Games\Need For Speed Hot Pursuit 2\memory(39,5): error X5335: Dest register cannot be the same as first source register for m*x* instructions.
```

The exact line that has the error is here:
```text
    m3x3 r3, r3, c5
```

Notice how both the destination and the first source register are both `r3`, which is not allowed in d3d9.

I created a patch for this.  You can download it here:

[d3d8.zip](https://github.com/crosire/d3d8to9/files/1465519/d3d8.zip)

**Comment by spartaque12** [comment]
elishacloud , thank u so much

**Comment by elishacloud** [comment]
This should be fixed with the latest [release v1.7.0](https://github.com/crosire/d3d8to9/releases/tag/v1.7.0).
